homeassistant/components/overkiz/sensor.py

```diff
@@ -46,11 +46,15 @@
 
 def _unit_of_measurement(device: Device, default: str) -> str | None:
     """Retrieve the unit of measurement from Overkiz attributes."""
-    if unit := device.attributes.get(OverkizAttribute.CORE_MEASURED_VALUE_TYPE):
-        if unit.value_as_str:
-            return OVERKIZ_UNIT_TO_HA.get(unit.value_as_str, default)
-        else:
-            return default
+    attrs = device.attributes
+    if (unit := attrs[OverkizAttribute.CORE_MEASURED_VALUE_TYPE]) and unit.value_as_str:
+        return OVERKIZ_UNIT_TO_HA.get(unit.value_as_str, default)
+    if (
+        unit := attrs[
+            OverkizAttribute.CORE_ELECTRIC_POWER_CONSUMPTION_STATE_MEASURED_VALUE_TYPE
```

Comment:
Should we apply this one only to the `OverkizState.ELECTRIC_POWER_CONSUMPTION_STATE`? We don't need this to broad.

Reply:
Ah yes you are right, I will add a check on the state name too for this one. Thx for the remark.

Reply:
Fixed.